packages/contracts/test/AZTEC/bilateralSwap.js

```diff
@@ -0,0 +1,156 @@
+/* global artifacts, expect, contract, beforeEach, it:true */
+// ### External Dependencies
+const BN = require('bn.js');
+const crypto = require('crypto');
+const { padLeft, sha3 } = require('web3-utils');
+
+
+// ### Internal Dependencies
+const aztec = require('aztec.js');
+const { params: { t2, GROUP_MODULUS } } = require('aztec.js');
+const { note: { utils: { toBytes32 } } } = require('aztec.js');
+const exceptions = require('../../utils/exceptions');
+
+// ### Artifacts
+const BilateralSwap = artifacts.require('../../contracts/AZTEC/BilateralSwap');
+const BilateralSwapInterface = artifacts.require('../../contracts/AZTEC/BilateralSwapInterface');
+
+BilateralSwap.abi = BilateralSwapInterface.abi;
+
+/*
+Outline of test:
+- This needs to be able to run on Truffle - to test the smart contract feature
+- Construct success states
+- Generate a proof using the BilateralSwap javascript proof generation code
+- Validate that it works, first using the javascript proof verification code
+- Validate that it works, using the bilateral swap smart construct
+*/
+
+contract('BilateralSwap', (accounts) => {
+    describe('success states', () => {
+        let bilateralSwap;
+        let testNotes;
+
+        beforeEach(async () => {
+            bilateralSwap = await BilateralSwap.new(accounts[0]);
+            const makerNoteValues = [10, 20];
+            const takerNoteValues = [10, 20];
```

Comment:
We should generate note values randomly distributed from [0, ..., K_MAX] (I need to update my tests in `aztec.js` to do this as well!)

Reply:
Done